src/main/java/net/neoforged/neoforge/common/util/GlobalVec3.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.util;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.Vec3;
+import org.joml.Vector3f;
+
+public class GlobalVec3 extends Vec3 {
+    public final Level level;
```

Comment:
Why make this class? And why does it store the `Level` instance directly? If this is supposed to be a counterpart to `GlobalPos`, shouldn't it store a `ResourceKey<Level>` instead? 

Additionally, if we're offering this as API, should we complete the overriding of all `Vec3` methods to return a `GlobalVec3` class instead?

Reply:
I couldn't decide between making this a super-universal class and a super-specific one. If you think the former makes sense, I'm happy to extend it that way.